langextract/__init__.py

```diff
@@ -196,10 +196,18 @@ def extract(
 
     # Currently only Gemini is supported
     if not api_key and language_model_type == inference.GeminiLanguageModel:
-      raise ValueError(
-          "API key must be provided for cloud-hosted models via the api_key"
-          " parameter or the LANGEXTRACT_API_KEY environment variable"
-      )
+      if not language_model_params.get('vertexai') or \
+          not language_model_params.get('project') or \
+          not language_model_params.get('location'):
+        raise ValueError(
+            "Gemini models require you to provide either:\n"
+            "  - an API key via the api_key parameter or the"
+            " LANGEXTRACT_API_KEY environment variable\n"
+            "  - language_model_params with:\n"
+            "    - vertex_ai = True\n"
+            "    - project\n"
+            "    - location"
+        )
```

Comment:
Wondering if it wouldn't be better to do this in an `except ValueError` block when instantiating `language_model_type` on line 228, since the `genai` lib is already designed to handle authentication issues, meaning we could reuse its message and just add information about `LANGEXTRACT_API_KEY`.